antidox/perspective.py

```diff
@@ -126,39 +130,51 @@ def main(argv):
   parser.add_argument('--sql_query', help='choose specifications for query search')
   parser.add_argument('--csv_file', help='input csv file')
   parser.add_argument('--content', help='specify a column in dataset to retreive data from')
+  parser.add_argument('--output', help='path for output file')
+  parser.add_argument('--suffix', help='output file suffix')
+  parser.add_argument('--project', help='project id for bigquery table')
   args = parser.parse_args(argv)
   apikey_data, perspective, dlp = get_client('api_key.json')
-  if args.csv_file:
-    text = pd.read_csv(args.csv_file)
-  if args.sql_query:
-    big_q = bigquery.Client.from_service_account_json('querykey.json')
-    text = use_query(args.content, args.sql_query, big_q)
-  pii_results = open("pii_results.txt", "w+")
-  toxicity_results = open("toxicity_results.txt", "w+")
-  print(big_q)
-
-  for line in text:
-    try:
-      print(line)
-      print('==============================================\n')
-      dlp_response = dlp_request(dlp, apikey_data, line)
-      perspective_response = perspective_request(perspective, line)
-      has_pii_bool, pii_type = contains_pii(dlp_response)
-      if has_pii_bool:
-        pii_results.write(str(line)+"\n"+'contains pii?'+"Yes"+"\n"
-                          +str(pii_type)+"\n"
-                          +"==============================================="+"\n")
-      if contains_toxicity(perspective_response):
-        toxicity_results.write(str(line)+"\n" +"contains TOXICITY?:"+
-                               "Yes"+"\n"+
-                               str(perspective_response['attributeScores']
-                                   ['TOXICITY']['summaryScore']['value'])+"\n"
-                               +"=========================================="+"\n")
-    except google_api_errors.HttpError as err:
-      print('error', err)
-  toxicity_results.close()
-  pii_results.close()
-    # print('dlp result:', json.dumps(dlp_response, indent=2))
-    # print ("contains_toxicity:", json.dumps(perspective_response, indent=2))
+  with beam.Pipeline(options=PipelineOptions()) as pipeline:
+    if args.csv_file:
+      comments = pipeline | 'ReadMyFile' >> beam.io.ReadFromText(args.csv_file)
+    # if args.sql_query:
```

Comment:
Can you keep this if statement, so that this code continues to work with CSV, BigQuery, and Wikipedia (once the recent changes are merged)

Reply:
don

Reply:
done